### ISSUE TITLE
Keypress being read by non focused windows

### ISSUE THREAD

**Issue by davidoskky** [i3/i3#3609]
## I'm submitting a…
<pre>
[x] Bug
[ ] Feature Request
[ ] Documentation Request
[ ] Other (Please describe in detail)
</pre>

## Current Behavior

The problem happens using the program rtv 1.24.0.
From the program I can open images, it launches a new window with the command `feh --scale-down --image-bg black <URL>`.
feh version 3.1.1
If after opening an image I focus back on rtv and push keys once in a while some of the keys get read from feh as well.

In particular jk are used in rtv to go up and down and I set them in feh to zoom in and out.
Once in a while (every 4-10 key presses, it's not consistent) instead of rtv going down I'll have feh zoom out.
I tried with other keys and the problem happens consistently.

Adding other windows or changing the splitting to horizontal doesn't change the problem

I tried using the default configs and submitted the log file, the problem still persists

## Expected Behavior

When I use a program I want the key presses to be read by that program only.

## Reproduction Instructions

Open rtv, open a feh window from rtv focus back on rtv and press keys which have a binding on feh.

## Environment
Output of `i3 --moreversion 2>&-`:
<pre>
Binary i3 version:  4.16.1 (2019-01-27) © 2009 Michael Stapelberg and contributors
Running i3 version: 4.16.1 (2019-01-27) (pid 7137)abort…)
Loaded i3 config: /home/davidoskky/.config/i3/config (Last modified: sab 02 feb 2019 12:44:10 CET, 2345 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
</pre>

<pre>
Logfile URL: https://logs.i3wm.org/logs/5751164012331008.bz2
</pre>

<pre>
- Linux Distribution & Version:Arch 4.20.6
- Are you using a compositor (e.g., xcompmgr or compton):Compton
</pre>

**Comment by davidoskky** [comment]
A small update: I tried the same using sxiv and the problem did not occur, it must be somehow feh specific.

**Comment by einsweniger** [comment]
I have a similar issue, but it's weird to reproduce as well.

Launching a terminal with i3 keybindings ($mod+Return) it will work just fine, but when I launch my programs through `dmenu`, then I have the same focus issue as OP. The new window is spawned with active focus, but the input is eaten by whatever is currently under the mouse.

All of this worked some time ago, I'll try and build older i3 versions to verify that it is really an i3 issue, since dmenu was updated at the same time (on my machine).

**Comment by einsweniger** [comment]
welp, I could not compile the previous version of i3, but the previous version of dmenu did compile: and it fixed it for me. Maybe checking the changes (v4.8 -> v4.9) might give some insight what went wrong?

**Comment by einsweniger** [comment]
might be this https://git.suckless.org/dmenu/commit/db6093f6ec1bb884f7540f2512935b5254750b30.html

> revert parts of commit 377bd37e212b1ec4c03a481245603c6560d0be22
> this commit also broke input focus.

**Comment by orestisf1993** [comment]
@einsweniger : https://github.com/i3/i3/issues/3528

**Comment by seidler2547** [comment]
Yep, same issue here, i3 4.17. When I start galculator through dmenu_run, it becomes a floating window (as per `for_window` and it has the focused border color, but the keyboard input goes to the previously focused window. Very easy to reproduce: default config, remove `exec i3-config-wizard`, add `for_window [instance="galculator"] floating enable` (or any other program if you like), start i3, hit Mod1+d for dmenu and start galculator. It won't accept keyboard input unless it's unfocused (e.g. by mouse out) and refocused (e..g by mouse in) or refocused via keyboard shortcut. This still works correctly in the Debian stretch version 4.13.

**Comment by rgcv** [comment]
Experiencing something along these lines on 4.18. When running something from `dmenu`, despite seeing the newly created window focused, the previously focused window is still capturing keyboard input.

Curiously enough, say the new window is open in a split to the right of the existing window. If the mouse is placed atop the soon-to-appear window, focus is now on the new window. Fiddling with the `focus-follows-mouse` setting proves futile as the issue remains.

Some of my system's details:

```sh
$ uname -srm
Linux 5.5.8-arch1-1 x86_64
$ i3 --moreversion
Binary i3 version:  4.18 (2020-02-17) © 2009 Michael Stapelberg and contributors
(Getting version from running i3, press ctrl-c to abort…)
Running i3 version: 4.18 (2020-02-17) (pid 2810)
Loaded i3 config: /home/rgcv/.config/i3/config (Last modified: Fri 21 Feb 2020 06:24:17 PM WET, 1285156 seconds ago)

The i3 binary you just called: /usr/bin/i3
The i3 binary you are running: i3
```

**Comment by muhmud** [comment]
I recently switched to st (within tabbed) and have now started to get this issue also (i3 4.19) - maybe suckless related somehow?

**Comment by humeniuc** [comment]
I have a simiar situation concerning `i3` and `tabbed` from suckless. I start `tabbed` with a few tabs (xterm or st) and I cannot change focus to another window.

The behavior can be observed in here: [https://www.youtube.com/watch?v=Dedbrw9WAa0](https://www.youtube.com/watch?v=Dedbrw9WAa0)

Steps to reproduce:
1. Create a new workspace.
2. Press `Win + d` to start `dmenu`. Type `scite`. Or any other program you could type into.
3. Press `Win + d` to start `dmenu`. Type `tabbed xterm -into`. The new window is opened to the right.
4. With `tabbed` window focused, press a few `Ctrl + Shift + Enter` to create new tabs in `tabbed`.
5. First bug: Press `Win + ←`. The left window does not receive focus. I have to repeat pressing `Win + ←` a few times to see the left (`scite`) window receiving focus.
6. Second bug: The left (`scite`) window is shown as highlighted/focused. But the keypress are registered in the `tabbed`/`xterm` window.

i3 -version is `4.20.1`  
The os is arch
I tested with the aur's `tabbed-git` version and with a locally compiled one. The behavior is the same.
